x-pack/plugins/ml/public/application/services/anomaly_explorer_charts_service.ts

```diff
@@ -0,0 +1,1060 @@
+        config.metricFunction = ES_AGGREGATION.SUM;
+        config.metricFieldName = summaryCountFieldName;
+      }
+    }
+
+    return config;
+  }
+
+  public buildConfig(record: ChartRecord, job: CombinedJob): SeriesConfigWithMetadata {
+    const detectorIndex = record.detector_index;
+    const config: Omit<
+      SeriesConfigWithMetadata,
+      'bucketSpanSeconds' | 'detectorLabel' | 'fieldName' | 'entityFields' | 'infoTooltip'
+    > = {
+      ...this.buildConfigFromDetector(job, detectorIndex),
+    };
+
+    const fullSeriesConfig: SeriesConfigWithMetadata = {
+      bucketSpanSeconds: 0,
+      entityFields: [],
+      fieldName: '',
+      // @ts-ignore
+      infoTooltip: {},
+      ...config,
+    };
+    // Add extra properties used by the explorer dashboard charts.
+    fullSeriesConfig.functionDescription = record.function_description;
+    fullSeriesConfig.bucketSpanSeconds = parseInterval(
+      job.analysis_config.bucket_span
+    )!.asSeconds();
+
+    fullSeriesConfig.detectorLabel = record.function;
+    const jobDetectors = job.analysis_config.detectors;
+    if (jobDetectors) {
+      fullSeriesConfig.detectorLabel = jobDetectors[detectorIndex].detector_description;
+    } else {
+      if (record.field_name !== undefined) {
+        fullSeriesConfig.detectorLabel += ` ${fullSeriesConfig.fieldName}`;
+      }
+    }
+
+    if (record.field_name !== undefined) {
+      fullSeriesConfig.fieldName = record.field_name;
+      fullSeriesConfig.metricFieldName = record.field_name;
+    }
+
+    // Add the 'entity_fields' i.e. the partition, by, over fields which
+    // define the metric series to be plotted.
+    fullSeriesConfig.entityFields = getEntityFieldList(record);
+
+    if (record.function === ML_JOB_AGGREGATION.METRIC) {
+      fullSeriesConfig.metricFunction = mlFunctionToESAggregation(record.function_description);
+    }
+
+    // Build the tooltip data for the chart info icon, showing further details on what is being plotted.
+    let functionLabel = `${config.metricFunction}`;
+    if (
+      fullSeriesConfig.metricFieldName !== undefined &&
+      fullSeriesConfig.metricFieldName !== null
+    ) {
+      functionLabel += ` ${fullSeriesConfig.metricFieldName}`;
+    }
+
+    fullSeriesConfig.infoTooltip = {
+      jobId: record.job_id,
+      aggregationInterval: fullSeriesConfig.interval,
+      chartFunction: functionLabel,
+      entityFields: fullSeriesConfig.entityFields.map((f) => ({
+        fieldName: f.fieldName,
+        fieldValue: f.fieldValue,
+      })),
+    };
+
+    return fullSeriesConfig;
+  }
+  public async getCombinedJobs(jobIds: string[]): Promise<CombinedJob[]> {
+    const jobs = await Promise.all(
+      // Getting only necessary job config and datafeed config without the stats
+      jobIds.map((jobId) => this.mlApiServices.jobs.jobForCloning(jobId))
+    );
+    const results = jobs
+      .filter(isDefined)
```

Comment:
I actually don't think isCombinedJob is good here because jobForCloning gives a pair of `{job, datafeed}` and not just `job`. This function is taking the response from `jobForCloning` and combining them to get the `CombinedJob` shape.